pkg/library/libraryinputresources/prune_mustgather.go

```diff
@@ -87,6 +87,18 @@ func GetRequiredInputResourcesForResourceList(ctx context.Context, resourceList
 		instances = append(instances, resourceInstance)
 	}
 
+	for _, currResource := range resourceList.LabelSelectedResources {
+		resourceList, err := getResourcesByLabelSelector(ctx, dynamicClient, currResource)
+		if apierrors.IsNotFound(err) {
+			continue
+		}
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		instances = append(instances, resourceList...)
```

Comment:
we need to eliminate duplicate adds to the `instances`.  Something can be explicitly listed AND referenced AND label selected, but it must only appear in the `instances` once

Reply:
Added a type to avoid duplicate adds